### DISCUSSION TITLE
problem with rag: TypeError: 'NoneType' object is not iterable

### DISCUSSION THREAD

**Discussion by mirkancal** [open-webui/open-webui#4619] [General]
# Bug Report

I have no problem with file upload but RAG never worked on my end. I didn't do anything else except uploading file and sending to chat, do I need to install some pipeline or tool for a basic rag?
<img width="571" alt="image" src="https://github.com/user-attachments/assets/1198d46c-93de-4371-9873-32491b0b340f">
<img width="609" alt="image" src="https://github.com/user-attachments/assets/c34a9a81-5da6-4eb5-b705-a0c016091a6c">

## Installation Method

Docker, Ubuntu server

## Environment

- **Open WebUI Version:** v0.3.13
[(latest)](https://github.com/open-webui/open-webui/releases/tag/v0.3.13)

- **Operating System:** :  Ubuntu 24.04 LTS

Docker version 27.1.2, build d01f264

- **Browser (if applicable):** Latest Arc Browser, trying on chrome and firefox as of now.

**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [X] I have included the Docker container logs.
- [X] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Basic embedding search.

## Actual Behavior:

No idea about file(expected) but also no idea about questions.

## Description

**Bug Summary:**
[Provide a brief but clear summary of the bug]

## Reproduction Details

**Steps to Reproduce:**
I'm using openai embeddings.
Uploading any pdf file and ask my questions.

## Logs and Screenshots

**Browser Console Logs:**
No network or console errors on devtools.

**Docker Container Logs:**
```
ERROR [apps.rag.main] 'NoneType' object is not iterable
Traceback (most recent call last):
  File "/app/backend/apps/rag/main.py", line 998, in store_docs_in_vector_db
    embeddings = embedding_func(embedding_texts)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/apps/rag/utils.py", line 236, in <lambda>
    return lambda query: generate_multiple(query, func)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/apps/rag/utils.py", line 229, in generate_multiple
    embeddings.extend(f(query[i : i + batch_size]))
TypeError: 'NoneType' object is not iterable
```

**Screenshots/Screen Recordings (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by c0sogi** [comment]
Me either. I think this is due to `429 client error` for big text query.
Refer doc for rate limit: https://platform.openai.com/docs/guides/rate-limits/usage-tiers?context=tier-one

and this might resolve your problem.
https://github.com/open-webui/open-webui/pull/4698

**Comment by silenceroom** [comment]
had the same problem, after investigation, I found that the reaon was due to:
1. The OpenAI Model allow list did not enavble the text-embeding models.
2. The text might be too large hence need to utlize the large model instead of hte default small model. 

after that the CPU resource might also be a obstacle as it comsmes lots of CPU/RAM while extracting unformatted text from the document file.

**Comment by WesleyBatista** [comment]
I was facing this `'NoneType' object is not iterable` toast on the UI no matter text file sizes I would upload.

Issue was gone after changing **Embedding Model Engine** from `Ollama` to `Default (SentenceTransformers)`.

I still wanted **Ollama** as the engine though, so I tried to change `Embedding Model` from `granite-embedding:278m` to `mxbai-embed-large` and it worked! fyi, these other ones also worked:
- nomic-embed-text
- bge-m3
- snowflake-arctic-embed

Not sure if there was some other setting making `granite-embedding:278m` unusable for me, but worth checking that out!

**Reply by WaldemarEnns** [reply]
For me, switching from either `ollama` or `OpenAI` to `Default (sentence-transformers)` fixed it. The OpenAI embedding model is allowed in my API settings and ollama is up and running, none of them works and I receive the same error in the frontend.

Not sure why that happens but for now, the Default model works.

**Reply by rgaricano** [reply]
just a question: when you change embedd model to ollamas or openai, do you use same embedd model as in default? and if you use different:  did you reset database?
